Makefile

```diff
@@ -76,7 +77,7 @@ ifeq ($(CONFIG_BUILD_KERNEL),y)
 
 install: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_install)
 
-$(BIN): $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_all)
+$(BIN): $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_romloader) $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_all)
```

Comment:
is it good design to modify the common script to support a special application needs?

Reply:
I don't like it either but it was the only way I found to make sure that the romloader app is compiled after all the required files are generated (without making it a kconfig dependency)

Reply:
Let's user add `ROMLOADER_COPY += xxx` to Make.defs, you can use ROMLOADER_COPY in your Makefile directly.

Reply:
This was what I did in one of my first attempts. It works fine for static files. The problem was when the files are generated or downloaded during the compilation process. I wasn't able to call these files recipes from the romloader Makefile. Is there a way around this ?

Reply:
In this case, the user could do the generation in context target just like how we deal with the 3rd party libraries. I found that context phase is a best place to resolve the cross project dependence.

Reply:
Doing something like ```context:: $(ROMLOADER_COPY)``` in the romloader makefile ? I tried this but it couldn't find the target for the files as it is in a different makefile.
I'm probably missing something. Idk a lot of how the NuttX build system works. Could you point me to an example of where this is done ?

Reply:
No, I mean the user could generate the romfs content in context phase:
```
I don't like it either but it was the only way I found to make sure that the romloader app is compiled after all the required files are generated (and without making it a kconfig dependency).
```